### ISSUE TITLE
When disconnecting BLE HID, it doesn't restart advertising

### ISSUE THREAD

**Issue by mondalaci** [UltimateHackingKeyboard/firmware#850]
After disconnecting my UHK 80 via BLE HID on any OS, advertising doesn't restart, and I get the following in the Zephyr console:

```
Disconnected from unknown (ac:d6:18:1b:3d:b5), reason 19
Advertising failed to start (err -12)
```

**Comment by mondalaci** [comment]
@benedekkupper Can you reproduce and fix this issue?

**Comment by benedekkupper** [comment]
This doesn't happen to me, but the error code for disconnecting at you also seems out of place.

**Comment by mondalaci** [comment]
@benedekkupper I can consistently reproduce this issue with Android, Windows, iOS, and macOS, so I can't imagine how you can't reproduce it. Please try to reproduce it again. The full Zephyr log I see after disconnection:

```
Disconnected from unknown (ac:d6:18:1b:3d:b5), reason 19
Advertising failed to start (err -12)
I: stopping HID app 1
I: stopping HID app 1
I: stopping HID app 1
I: stopping HID app 1
I: stopping HID app 1
```

I forgot to paste the `I: stopping HID app 1` messages in the original description, but they show up multiple times.

**Comment by mondalaci** [comment]
@benedekkupper, Can you prioritize this unless you're working on another issue? Can you reproduce it?

**Comment by benedekkupper** [comment]
I haven't been able to reproduce this still. My biggest problem is that at this point most of my devices aren't able to connect to the UHK over BLE anymore, so that's what I must deal with first.

**Comment by mondalaci** [comment]
Any ideas why? On the UHK side, deleting settings partition keys one by one manually or via `nrfjprog --recover` should work. On the host OS side, solutions are OS-specific.

**Comment by benedekkupper** [comment]
> The UHK 80 is only visible from Android now, even though other OSes detected it earlier. After git-bisecting the issue, I found that https://github.com/UltimateHackingKeyboard/firmware-uhk80/commit/30efd8a880d63888551756d5036c2f21dff9e88e#diff-ced8391d5849741d811adc59c5b005f9d7f16ba9f1409fbc31ca4a6e69b35eafR75 is the culprit. I tried to raise CONFIG_BT_MAX_CONN from 2 to 3 to no avail. Any ideas?

Here is the documentation of the BLE advertisement function:
```
/**
 * @brief Start advertising
 *
 * Set advertisement data, scan response data, advertisement parameters
 * and start advertising.
 *
 * When the advertisement parameter peer address has been set the advertising
 * will be directed to the peer. In this case advertisement data and scan
 * response data parameters are ignored. If the mode is high duty cycle
 * the timeout will be @ref BT_GAP_ADV_HIGH_DUTY_CYCLE_MAX_TIMEOUT.
 *
 * @param param Advertising parameters.
 * @param ad Data to be used in advertisement packets.
 * @param ad_len Number of elements in ad
 * @param sd Data to be used in scan response packets.
 * @param sd_len Number of elements in sd
 *
 * @return Zero on success or (negative) error code otherwise.
 * @return -ENOMEM No free connection objects available for connectable
 *                 advertiser.
 * @return -ECONNREFUSED When connectable advertising is requested and there
 *                       is already maximum number of connections established
 *                       in the controller.
 *                       This error code is only guaranteed when using Zephyr
 *                       controller, for other controllers code returned in
 *                       this case may be -EIO.
 */
```
You are getting `-12 == -ENOMEM` error, which means:

> No free connection objects available for connectable advertiser.

From that I'd take that indeed you will need to extend the number of supported connections. Why 3 wasn't sufficient is strange indeed, I will need to debug this method to see which condition exactly is giving the error.

**Comment by benedekkupper** [comment]
Another related issue that I observed with macOS just now: if the NUS advertisement is also active (so the left is not connected via BLE to the right), then the PC doesn't "see" the HID advertisement of the right side.

**Comment by mondalaci** [comment]
@benedekkupper I've fixed this issue via e44d01d0b0e5178030cc4c2c696e3a31191a31c8, but upon firmware startup, I get "NUS and HID advertising continued" instead of "NUS and HID advertising successfully started", which makes me feel uneasy. Please look into this.
